src/SURFnet/OATHBundle/Services/UserStorage/Encryption/Mcrypt.php

```diff
@@ -0,0 +1,59 @@
+<?php 
+
+namespace SURFnet\OATHBundle\Services\UserStorage\Encryption;
+
+/**
+ * Class for encrypting/decrypting the user secret with mcrypt.
+ * 
+ * @author peter
+ */
+class Mcrypt implements UserEncryptionInterface
+{
+    private $_cipher;
+    private $_mode;
+    private $_key;
+    private $_iv;
+    
+    /**
+     * Construct an encryption instance.
+     *
+     * @param $config The configuration that a specific configuration class may use.
+     */
+    public function __construct($config)
+    {
+        $this->_cipher = $config['cipher'];
+        $this->_mode = $config['mode'];
+        $this->_key = $config['key'];
+
+        if (!empty($config['iv'])) {
+            $this->_iv = $config['iv'];
+        } else {
+            $iv_size = mcrypt_get_iv_size($this->_cipher, $this->_mode);
+            $this->_iv = mcrypt_create_iv($iv_size, MCRYPT_RAND);
```

Comment:
A new iv is created for every for every instance. That's fine, but you should then store the iv, otherwise there is no way to decrypt secrets.